src/upload.ts

```diff
@@ -195,3 +227,25 @@ export async function uploadDirectoryRequest(
 function ensureFileObjectConsistency(file: File): File {
   return new File([file], file.name, { type: getFileMimeType(file) });
 }
+
+function validateFile(name: string, value: unknown, valueKind: string) {
+  if (!(value instanceof File)) {
+    throwValidationError(name, value, valueKind, "'File'");
+  }
+}
+
+function validateUploadResponse(response: AxiosResponse): void {
+  try {
+    if (!response.data) {
+      throw new Error("response.data field missing");
+    }
+
+    validateString("response.data.skylink", response.data.skylink, "upload response field");
```

Comment:
not sure these 3 instances warranted the abstraction where you need to pass `parameter` or `upload reponse field` - but I don't mind I think.

I wouldn't mention `response.data` in the field name though. Isn't that weird? I think `response.data` is kind of implied by `upload response field`.